hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -57,13 +57,23 @@
   public static final String SUFFIX_READ_OPTIMIZED_TABLE = "_ro";
 
   private final HiveSyncConfig cfg;
-  private final HoodieHiveClient hoodieHiveClient;
+  private HoodieHiveClient hoodieHiveClient = null;
   private final String snapshotTableName;
   private final Option<String> roTableTableName;
 
   public HiveSyncTool(HiveSyncConfig cfg, HiveConf configuration, FileSystem fs) {
     super(configuration.getAllProperties(), fs);
-    this.hoodieHiveClient = new HoodieHiveClient(cfg, configuration, fs);
+
+    try {
+      this.hoodieHiveClient = new HoodieHiveClient(cfg, configuration, fs);
```

Comment:
I guess there could be gap/bug in here. Did you happen to test it out. if hoodieHiveClient is null (due to connect execution) and if caller subsequently calls syncHoodieTable(), won't we end up with NullPointerException? Can you please check.